h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java

```diff
@@ -88,12 +86,13 @@ private ValNum scalar(Frame frx, Frame fry, Mode mode) {
     for (int r = 0; r < ncols; r++) {
       xval = vecxs[r].at(0);
       yval = vecys[r].at(0);
-      if (!(Double.isNaN(xval) || Double.isNaN(yval)))
+      if (!(Double.isNaN(xval) || Double.isNaN(yval))) {
         //Compute variance of x and y vars
         xvar += Math.pow((vecxs[r].at(0) - xmean), 2);
-      yvar += Math.pow((vecys[r].at(0) - ymean), 2);
-      //Compute sum of squares of x and y
-      ss += (vecxs[r].at(0) - xmean) * (vecys[r].at(0) - ymean);
+        yvar += Math.pow((vecys[r].at(0) - ymean), 2);
```

Comment:
Not sure if it worked correctly before. New test catches all permutation of this method.